src/modules/FioAddress/util.js

```diff
@@ -328,3 +328,69 @@ export const checkPubAddress = async (fioPlugin: EdgeCurrencyConfig, fioAddress:
 
   return ''
 }
+
+export const checkRecordSendFee = async (fioWallet: EdgeCurrencyWallet, fioAddress: string) => {
+  try {
+    const getFeeResult = await fioWallet.otherMethods.fioAction('getFee', {
+      endPoint: 'record_obt_data',
+      fioAddress: fioAddress
+    })
+    if (getFeeResult.fee) {
+      throw new Error(s.strings.fio_no_bundled_err_msg)
+    }
+  } catch (e) {
+    throw new Error(s.strings.fio_get_fee_err_msg)
+  }
+}
+
+export const recordSend = async (
+  senderWallet: EdgeCurrencyWallet,
+  senderFioAddress: string,
+  params: {
+    payeeFioAddress: string,
+    payerPublicAddress: string,
+    payeePublicAddress: string,
+    amount: string,
+    currencyCode: string,
+    chainCode: string,
+    txid: string,
+    memo: string
+  }
+) => {
+  if (senderFioAddress && senderWallet) {
+    const { payeeFioAddress, payerPublicAddress, payeePublicAddress, amount, currencyCode, chainCode, txid, memo } = params
+    try {
+      await senderWallet.otherMethods.fioAction('recordObtData', {
+        payerFioAddress: senderFioAddress,
+        payeeFioAddress,
+        payerPublicAddress,
+        payeePublicAddress,
+        amount,
+        tokenCode: currencyCode,
+        chainCode,
+        obtId: txid,
+        memo,
+        maxFee: 0,
+        tpid: '',
+        status: 'sent_to_blockchain'
+      })
+    } catch (e) {
+      //
+      throw new Error(e.message)
+    }
+  }
+}
+
+export const getFioObtData = async (fioWallets: EdgeCurrencyWallet[]): Promise<FioObtRecord[]> => {
```

Comment:
This can explode memory usage. Please change to have the FIO plugin only return OBT data for the requested transactions. This can be done in a future pr.

Reply:
ok. would it any difference of memory usage between getting data in plugin and in gui?